src/pages/Login.js

```diff
@@ -1,5 +1,119 @@
+import React from "react";
+import { Navigate, Link } from "react-router-dom";
+import { useMutation } from "react-query";
+import { useForm } from "react-hook-form";
+import {
+  TextField,
+  Button,
+  Typography,
+  CircularProgress,
+  Alert,
+  Box,
+  FormControl,
+} from "@mui/material";
+import { authSignInWith, authSignInWithGoogle } from "../firebase/auth";
+import { useAuth } from "../context/authContext";
+
 const Login = () => {
-  return <div>This is a Login page Test!</div>;
+  const { isLoggedIn } = useAuth();
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+  } = useForm();
+
+  const signInMutation = useMutation(authSignInWith, {
+    onError: (error) => {
+      console.error("Error signing in:", error);
+    },
+  });
+
+  const googleSignInMutation = useMutation(authSignInWithGoogle, {
+    onError: (error) => {
+      console.error("Error signing in with Google:", error);
+    },
+  });
+
+  const onSubmit = async (data) => {
+    const { email, password } = data;
+    try {
+      await signInMutation.mutateAsync({ email, password });
+    } catch (error) {
+      console.error("Error signing in:", error);
+    }
+  };
```

Comment:
Do you need an async operation here? You already handled the error case in the mutation configuration. Try catch doesn't look necessary.

```suggestion
  const onSubmit = (data) => {
    const { email, password } = data;
    signInMutation.mutateAsync({ email, password });
  };
```

Reply:
const onSubmit = (data) => {
    const { email, password } = data;
    signInMutation.mutateAsync({ email, password });
  };